pop-benchmarker/src/main/java/com/microsoft/testing/popbenchmarker/MainActivity.java

```diff
@@ -0,0 +1,385 @@
+            keyStore.load(null);
+
+            final KeyStore.Entry entry = keyStore.getEntry(KEYSTORE_ALIAS, null);
+            final PrivateKey privateKey = ((KeyStore.PrivateKeyEntry) entry).getPrivateKey();
+
+            final RSASSASigner signer = new RSASSASigner(privateKey);
+
+            final SignedJWT signedJWT = new SignedJWT(
+                    new JWSHeader.Builder(JWSAlgorithm.RS256)
+                            //.keyID(key.getKeyID())
+                            .build(),
+                    claimsSet
+            );
+
+            final Timer.TimerResult<Void> result = Timer.execute(new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    // Compute the RSA signature
+                    signedJWT.sign(signer);
+
+                    // To serialize to compact form, produces something like
+                    // eyJhbGciOiJSUzI1NiJ9.SW4gUlNBIHdlIHRydXN0IQ.IRMQENi4nJyp4er2L
+                    // mZq3ivwoAjqa1uUkSBKFIX7ATndFF5ivnt-m8uApHO4kfIFOrW7w2Ezmlg3Qd
+                    // maXlS9DhN0nUk_hGI3amEjkKd0BWYCB8vfUbUv0XGjQip78AI4z1PrFRNidm7
+                    // -jPDm5Iq0SZnjKjCNS5Q15fokXZc8u0A
+                    final String signed = signedJWT.serialize();
+                    //
+                    return null;
+                }
+            });
+            mSigningTimings.add(result.mDuration);
+            mThreadMarshaller.postResult(String.valueOf(result.mDuration), callback);
+        } catch (Exception e) {
+            crash(e);
+        }
+    }
+
+    private void getKeyLoadTiming(@NonNull final AsyncResultCallback<String> callback) {
+        final Timer.TimerResult<Void> result = Timer.execute(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                final KeyStore keyStore = KeyStore.getInstance(ANDROID_KEYSTORE);
+                keyStore.load(null);
+                keyStore.getEntry(KEYSTORE_ALIAS, null);
+                return null;
+            }
+        });
+        mKeyLoadTimings.add(result.mDuration);
+        mThreadMarshaller.postResult(String.valueOf(result.mDuration), callback);
+    }
+
+    private void getKeyGenerationTiming(@NonNull final AsyncResultCallback<String> callback) {
+        final Timer.TimerResult<Void> result = Timer.execute(new Callable<Void>() {
+            @Override
+            public Void call() {
+                KeyPairGenerator kpg;
+
+                try {
+                    while (true) {
+                        kpg = getInitializedRsaKeyPairGenerator(MainActivity.this);
+                        final KeyPair kp = kpg.generateKeyPair();
+                        // Check that the generated thumbprint size is 2048, if it is not, retry...
+                        final int length = RSAKeyUtils.keyBitLength(kp.getPrivate());
+
+                        if (length >= 2048 || length < 0) {
```

Comment:
Why check for greater than 2048 or less than 0? Why not just for exact 2048 as your comment above on line 336 suggests

Reply:
Private key material isn't exposed and so has a length of `-1` indicating 'hidden'